### DISCUSSION TITLE
SlidingWindow function on data frame that is indexed by date

### DISCUSSION THREAD

**Discussion by qwertz747** [timeseriesAI/tsai#351] [Q&A] [answered]
Hi,

thx a lot for this great work.
I am having a problem understanding the SlidingWindow Function in regard to my time series (which contains one entry per day and the index is the date, ascending).

I am using the following code for creating the X,y, and split.
![image](https://user-images.githubusercontent.com/36340314/147823605-f3204965-6ed6-4350-b87b-40929fb9a398.png)

It seems that the resulting shape is 351 long (and is missing the 14 most recent daily entries, not the **oldest** ones).
Is there a certain order I have to make sure the dataframe has to have when calling the slidingwindow function (ascending vs descending by date)?

![image](https://user-images.githubusercontent.com/36340314/147823490-69beb211-9638-421e-b92d-8205ff9d8a31.png)

**Comment by vrodriguezf** [comment]
The implementation of SlidingWindow has a `sort_by` and `ascending` arguments that you can play with. You may need the date to be a column itself, not an index.

**Comment by qwertz747** [comment, marked as answer]
Hi thx a lot for your reply @vrodriguezf . 
It got me thinking/experimenting and my error came from recreating the test and train dataframe manually not from the slidingwindow function. 

So the slidingwindow function was working as expected all the time!

This is how I manually recreated the test and train data frame correctly... if anyone has a similar issue... 
<img width="593" alt="image" src="https://user-images.githubusercontent.com/36340314/147848800-0805d2a2-1618-4229-83a0-bb73b87862f9.png">

Thx for the fast help!
